modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxRecoveryManager.java

```diff
@@ -0,0 +1,130 @@
+import org.apache.ignite.internal.logger.IgniteLogger;
+import org.apache.ignite.internal.logger.Loggers;
+import org.apache.ignite.internal.manager.IgniteComponent;
+import org.apache.ignite.internal.tx.LockManager;
+import org.apache.ignite.internal.tx.TxManager;
+import org.apache.ignite.internal.tx.TxState;
+import org.apache.ignite.internal.tx.TxStateMeta;
+import org.apache.ignite.internal.tx.event.LockEvent;
+import org.apache.ignite.internal.tx.event.LockEventParameters;
+import org.apache.ignite.internal.tx.message.TxMessageGroup;
+import org.apache.ignite.internal.tx.message.TxMessagesFactory;
+import org.apache.ignite.internal.tx.message.TxRecoveryMessage;
+import org.apache.ignite.network.ClusterNode;
+import org.apache.ignite.network.ClusterService;
+
+/**
+ * The manager sends and receives transaction recovery messages.
+ * <ul>
+ * <li>The recovery messages are sent when a lock conflict is found and the transaction coordinator is not available.</li>
+ * <li>The recovery message is received when the node has a commit partition for the specific transaction.</li>
+ * </ul>
+ */
+public class TxRecoveryManager implements IgniteComponent {
+    /** The logger. */
+    private static final IgniteLogger LOG = Loggers.forClass(TxRecoveryManager.class);
+
+    /** Tx messages factory. */
+    private static final TxMessagesFactory FACTORY = new TxMessagesFactory();
+
+    /** Lock manager. */
+    private final LockManager lockManager;
+
+    /** Transaction manager. */
+    private final TxManager txManager;
+
+    /** Cluster service. */
+    private final ClusterService clusterService;
+
+    /** The listener tries to clarify the holder transaction state to avoid permanent locking. */
+    private final EventListener<LockEventParameters> lockConflictListener;
+
+    /**
+     * The constructor.
+     *
+     * @param lockManager Lock manager.
+     * @param txManager Transaction manager.
+     * @param clusterService Cluster service.
+     */
+    public TxRecoveryManager(
+            LockManager lockManager,
+            TxManager txManager,
+            ClusterService clusterService
+    ) {
+        this.lockManager = lockManager;
+        this.txManager = txManager;
+        this.clusterService = clusterService;
+
+        lockConflictListener = (parameters, exception) -> {
+            TxStateMeta txSate = txManager.stateMeta(parameters.getLockHolderTx());
+
+            if (txSate == null || clusterService.topologyService().getById(txSate.txCoordinatorId()) == null) {
+                LOG.info("Conflict was found, and the coordinator of the transaction that holds a lock is not available "
+                                + "[lockHolderTx={}, lockAcquirerTx={}].",
+                        parameters.getLockHolderTx(),
+                        parameters.getLockAcquirerTx()
+                );
+
+                //TODO:IGNITE-20735 Obviously, the commit partition node does not have to be at the local node.
```

Comment:
The ticket says:

`Transactions X that detects (detection logic will be covered in a separate ticket) that coordinator is dead awaits commitPartition primary replica and sends initiateRecoveryReplicaRequest to it in a fully asynchronous manner`

Why do you just pick a local node? IMO the part of sending the initiate recovery message should be fully covered by this PR. The mentioned ticket IGNITE-20735 hasn't anything in description about this part of work.